Zodiac Goodie

n0mady opened the pull request:
What does your instant answer do?
Calculate the Zodiac/Star Sign for User Provided Date

What problem does your instant answer solve (Why is it better than organic links)?
Provides the Zodiac Sign for the User

What is the data source for your instant answer? (Provide a link if possible)
Wiki : https://en.wikipedia.org/wiki/Zodiac

Why did you choose this data source?
Reliable Resource

Are there any other alternative (better) data sources?
No

What are some example queries that trigger this instant answer?
zodiac 15th Feb
10 March 1789 starsign
7th Jan Star Sign

Which communities will this instant answer be especially useful for? (gamers, book lovers, etc)
General User especially people who believe in Horoscopes

Is this instant answer connected to a DuckDuckHack instant answer idea?
No.But this question is posted for clarifications of development
https://duck.co/forum/thread/5725/zodiac-starsigns

Which existing instant answers will this one supercede/overlap with?
No Existing Answers

Are you having any problems? Do you need our help with anything?
As of Now,No Thanks!!!
Cheers!!!

**Note: Please attach a screenshot for new instant answer pull requests, and for pull requests which modify the look/design of existing instant answers.

## Checklist
Please place an 'X' where appropriate.

[X] Added metadata and attribution information
[X] Wrote test file and added to t/ directory
[X] Verified that instant answer adheres to design guidelines(https://github.com/duckduckgo/duckduckgo-documentation/blob/master/duckduckhack/styleguides/design_styleguide.md)
[I have only Linux Installed in my System.I tested it in Firefox and Konqueror.Regrets] Tested cross-browser compatability


    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [] Google Chrome   
        [] Firefox         
        [] Opera           
        [] IE 10           

    - Windows 7
        [] Google Chrome   
        [] Firefox         
        [] Opera           
        [] IE 8            
        [] IE 9            
        [] IE 10           

    - Windows XP
        [] IE 7            
        [] IE 8            

    - Mac OSX
        [] Google Chrome   
        [] Firefox         
        [] Opera           
        [] Safari          

    - iOS (iPhone)
        [] Safari Mobile   
        [] Google Chrome   
        [] Opera           

    - iOS (iPad)
        [] Safari Mobile   
        [] Google Chrome   
        [] Opera            

    - Android
        [] Firefox         
        [] Native Browser  
        [] Google Chrome   
        [] Opera

jagtalon commented:
@mintsoft made a tutorial for this. Check out https://duck.co/duckduckhack/goodie_advanced_handle_functions#using-structured-responses!

javathunderman commented:
@jagtalon @mintsoft I thought I saw something like this. Is there another goodie like this one?

moollaza commented:
@javathunderman are you thinking about the ChineseZodiac Goodie? That is slightly different because it gives the Chinese Zodiac animal for your birth year. This gives the "Western" Zodiac sign, e.g. Scorpio, Capricorn, etc

javathunderman commented:
@moollaza Oh, that’s the one…:facepalm:

javathunderman commented:
@mintsoft We have to clean up documentation and the output format, and this can be merged, no?

mintsoft commented:
@javathunderman @n0mady It just needs switching to a structured answer and rebasing (there are merge conflicts) then it can be merged

jagtalon commented:
@javathunderman would you like to continue this? :)

n0mady commented:
Hi,
  I dont know if i have rebased the code properly because i have nevre done that.
  All I tried is merging the branches and avoiding the merge conflicts.
  I have modified the output format.
  Please let me know if anything to be done form my side.
  thanks for the support!

Cheers
:-)

javathunderman commented:
Checking with @n0mady before I go ahead. 

cc @jagtalon

mintsoft commented:
@n0mady it appears that you've removed the use of the date role in favour of `Date::Manip qw(ParseDate);` I think the rebase may have done this?

mintsoft commented:
@n0mady I think it was when this commit was resolved: https://github.com/n0mady/zeroclickinfo-goodies/commit/cae6bf07b66be40d7e1b7c0dd50bfc582aeaafb5

n0mady commented:
@mintsoft : Correct me if i am wrong. I still the line containing the ParseDate function in the commit
https://github.com/n0mady/zeroclickinfo-goodies/commit/e91753590486af36b6c938db6df70c9be32c2a6b
![selection_002](https://cloud.githubusercontent.com/assets/7482165/5991268/73082d78-aa08-11e4-82ad-056f5b8e581d.png)

n0mady commented:
Let me know if anything to be done from my side.

Cheers
:-)

mintsoft commented:
@n0mady looking at the changes that would be merged from this (https://github.com/duckduckgo/zeroclickinfo-goodies/pull/440/files) dist.ini looks conflicted and this isn't using the datesrole (https://github.com/duckduckgo/zeroclickinfo-goodies/pull/440/files#diff-1863d18c0094c366ace33a39253e7d38R11) any more. However it was using it before the merge conflicts were resolved?

Can you put it back to as it was in : https://github.com/n0mady/zeroclickinfo-goodies/commit/be613e97ac0dcd9c0224c98e61d85adb185d5313 but with the structured answer output?

n0mady commented:
I have added Date::Manip to the dist.ini file. Is this what you wanted or something else?
Did i understood correctly?

Cheers
:-)

mintsoft commented:
@n0mady That's not what I meant :) 

Before you merged and resolved the conflicts, the date parsing was handled by the DDG DatesRole that mwmiller fixed up (https://github.com/duckduckgo/zeroclickinfo-goodies/pull/440#issuecomment-54326461): 

https://github.com/n0mady/zeroclickinfo-goodies/commit/57ec1fd4de6f5011b825adabfb7dc22d7f7653c8

This has been completely changed by the rebasing, Is that what you intended?

n0mady commented:
updated

mintsoft commented on dist.ini:
```diff
@@ -75,9 +75,13 @@ Telephony::CountryDialingCodes = 1.04
 URI::Escape::XS = 0.12
 DateTime::Calendar::Chinese = 1.00
 DateTime::Event::Chinese = 1.00
+<<<<<<< HEAD
```
@n0mady Can you resolve these markings? I'm pretty sure these are why the tests are failing

mintsoft commented on dist.ini.orig:
```diff
@@ -0,0 +1,146 @@
+name    = DDG-GoodieBundle-OpenSourceDuckDuckGo
```
Can you delete this file (`dist.ini.orig`) as it's not needed

mintsoft commented:
@n0mady Excellent thanks for that! I've just spotted 2 other little issues that are getting in the way of the tests working. I think once they're fixed it'll run through Travis correctly :+1:

javathunderman commented:
@jagtalon Leaving in the hands of @n0mady then. 

cc @mintsoft

mintsoft commented:
Sorry @javathunderman !

javathunderman commented:
@mintsoft No problem, I would have messed this up anyway!

n0mady commented:
Changes pushed to the repo!

@javathunderman Thanks a lot for coming up for maintaining this. As I was under some medical attention in the past I was unable to continue this. Heartfelt thanks and apologies that I  came back and screwed your wish.

n0mady commented:
@javathunderman I am sure that you will someday become the future maintainer of this goodie if you wish if this goodie goes live. Because I am better in messing than you! :-)

mintsoft commented on lib/DDG/Goodie/Zodiac.pm:
```diff
@@ -0,0 +1,49 @@
+	input	=> [html_enc[date_output_string($zodiacdate).": "]],
+	operation => 'Star Sign for ',
+	result	=> html_enc(ucfirst(zodiac_date_name($zodiacdate)))
+	};
```
Can you remove this hard-tab here and replace with 4 spaces?

mintsoft commented:
@n0mady the tests are running now! However it looks like they've all failed: https://travis-ci.org/duckduckgo/zeroclickinfo-goodies/jobs/49129041#L1831

n0mady commented:
@mintsoft passed :-)

mintsoft commented on lib/DDG/Goodie/Zodiac.pm:
```diff
@@ -0,0 +1,52 @@
+    my $result="Star Sign for ".date_output_string($zodiacdate).": ".ucfirst(zodiac_date_name($zodiacdate));
+
+	#Input String
+	my $input=date_output_string($zodiacdate).": ";
```
Can you fix the indentation here? it should be 4 spaces

n0mady commented:
@mintsoft : Fixed

Cheers,
:-)

mintsoft commented:
@jagtalon @moollaza I think this is ready! :dancers:  Anything to add?

javathunderman commented:
@mintsoft @n0mady Good to see this going up!

jagtalon commented:
Sweet!!

jagtalon commented:
:+1:

mintsoft commented:
@n0mady Congratulations! :cake: This was pretty epic to get committed in :tada:

n0mady commented:
@jagtalon @mintsoft @javathunderman Thanks Everyone! Love you all! Will be happy to see my snippet live..:-)

moollaza commented:
@n0mady congrats! It's live: https://duckduckgo.com/?q=zodiac+15th+Feb&ia=answer

Sorry for the delay, its been live for about a week!

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's & Issues or even submit more Instant Answers!

javathunderman commented:
@moollaza Congrats!
